sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/QueueServiceClientProvider.cs

```diff
@@ -43,28 +43,20 @@ internal class QueueServiceClientProvider : StorageClientProvider<QueueServiceCl
         protected override QueueClientOptions CreateClientOptions(IConfiguration configuration)
         {
             var options = base.CreateClientOptions(configuration);
+
             options.MessageEncoding = _queuesOptions.MessageEncoding;
             return options;
         }
 
-        protected override QueueServiceClient CreateClientFromConnectionString(string connectionString, QueueClientOptions options)
+        protected override QueueServiceClient CreateClient(IConfiguration configuration, TokenCredential tokenCredential, QueueClientOptions options)
         {
             var originalEncoding = options.MessageEncoding;
             options.MessageEncoding = QueueMessageEncoding.None;
-            var nonEncodingClient = new QueueServiceClient(connectionString, options);
+            var nonEncodingClient = base.CreateClient(configuration, tokenCredential, options);
             options.MessageDecodingFailed += CreateMessageDecodingFailedHandler(nonEncodingClient);
             options.MessageEncoding = originalEncoding;
-            return new QueueServiceClient(connectionString, options);
-        }
 
```

Comment:
how is this going to work if connection string is used? I.e. how all these encoding related things are going to hookup now in that case?

Reply:
Same as before. `IConfiguration configuration` can represent both a connection string or a section so this method is a replacement for the two we had before.

This line would create a client using the connection string or the serviceUri but without the encoding.

```
            var nonEncodingClient = base.CreateClient(configuration, tokenCredential, options);

```
and this one would create the main one:

```
            return base.CreateClient(configuration, tokenCredential, options);
```